engine/testabilities/testmode/db_mode.go

```diff
@@ -48,5 +95,122 @@ func CheckPostgresMode() (ok bool, dbName string) {
 
 // CheckFileSQLiteMode checks if the test mode is set to use SQLite file
 func CheckFileSQLiteMode() bool {
-	return os.Getenv(modeEnvVar) == "file"
+	return os.Getenv(EnvDBMode) == "file"
+}
+
+// TestContainer represents a running test container
+type TestContainer struct {
+	Container testcontainers.Container
+	Host      string
+	Port      string
+}
+
+// GetOrCreatePostgres returns an existing PostgreSQL container or creates a new one if none exists
+// This helps reuse containers between tests
+func GetOrCreatePostgres(t testing.TB) *TestContainer {
+	t.Helper()
+
+	containerMutex.Lock()
+	defer containerMutex.Unlock()
+
+	if sharedContainer != nil {
+		t.Log("Reusing existing PostgreSQL container")
+		return sharedContainer
+	}
+
+	container := startPostgresContainer(t)
+	sharedContainer = container
+
+	if !ShouldSkipCleanup() {
+		t.Cleanup(func() {
+			containerMutex.Lock()
+			defer containerMutex.Unlock()
+
+			if sharedContainer != nil {
+				t.Log("Cleaning up PostgreSQL container")
+				ctx := context.Background()
+				if err := sharedContainer.Container.Terminate(ctx); err != nil {
+					t.Logf("Error terminating container: %v", err)
+				}
+				sharedContainer = nil
+			}
+		})
+	}
+
+	return container
+}
+
+// startPostgresContainer starts a PostgreSQL container
+func startPostgresContainer(t testing.TB) *TestContainer {
+	t.Helper()
+
+	skipCleanup := ShouldSkipCleanup()
+	if skipCleanup {
+		t.Log("Container cleanup will be skipped (EnvSkipCleanup=true)")
+	}
+
+	ctx := context.Background()
+
+	req := testcontainers.ContainerRequest{
+		Image:        "postgres:14",
```

Comment:
why not latest image?
I'm pretty sure, that nobody will remember to bump this version, especially as there is already version 17